base/workflow/R/run.write.configs.R

```diff
@@ -34,68 +35,82 @@ run.write.configs <- function(settings, write = TRUE, ens.sample.method = "unifo
-        #TODO: sometimes `files` is a 0x0 tibble and other operations with it fail.
-        files <- PEcAn.DB::dbfile.check("Posterior",
-                                        settings$pfts[[i]]$posteriorid, 
-                                        con, settings$host$name, return.all = TRUE)
-        pid <- grep("post.distns.*Rdata", files$file_name)  ## is there a posterior file?
-        if (length(pid) == 0) {
-          pid <- grep("prior.distns.Rdata", files$file_name)  ## is there a prior file?
-        }
-        if (length(pid) > 0) {
-          posterior.files[i] <- file.path(files$file_path[pid], files$file_name[pid])
-        }  ## otherwise leave posteriors as NA
-      }
-      ## otherwise leave NA and get.parameter.samples will look for local
-    } else {
-      ## does posterior.files point to a directory instead of a file?
-      if(utils::file_test("-d",posterior.files[i])){
-        pfiles = dir(posterior.files[i],pattern = "post.distns.*Rdata",full.names = TRUE)
-        if(length(pfiles)>1){
-          pid = grep("post.distns.Rdata",pfiles)
-          if(length(pid > 0)){
-            pfiles = pfiles[grep("post.distns.Rdata",pfiles)]
-          } else {
-            PEcAn.logger::logger.error(
-              "run.write.configs: could uniquely identify posterior files within",
-              posterior.files[i])
-          }
-          posterior.files[i] = pfiles
-        }
-      }
-      ## also, double check PFT outdir exists
-      if (is.null(settings$pfts[[i]]$outdir) || is.na(settings$pfts[[i]]$outdir)) {
-        ## no outdir
-        settings$pfts[[i]]$outdir <- file.path(settings$outdir, "pfts", settings$pfts[[i]]$name)
-      }
-    }  ## end else
-  } ## end for loop over pfts
-  
-  ## Sample parameters
   model <- settings$model$type
   scipen <- getOption("scipen")
   options(scipen = 12)
 
-  PEcAn.uncertainty::get.parameter.samples(settings, posterior.files, ens.sample.method)
-  samples.file <- file.path(settings$outdir, "samples.Rdata")
-  if (file.exists(samples.file)) {
-    samples <- new.env()
-    load(samples.file, envir = samples) ## loads ensemble.samples, trait.samples, sa.samples, runs.samples, env.samples
-    trait.samples <- samples$trait.samples
-    ensemble.samples <- samples$ensemble.samples
-    sa.samples <- samples$sa.samples
-    runs.samples <- samples$runs.samples
-    ## env.samples <- samples$env.samples
+#  PEcAn.uncertainty::get.parameter.samples(settings, posterior.files, ens.sample.method)
+#  samples.file <- file.path(settings$outdir, "samples.Rdata")
+#  if (file.exists(samples.file)) {
+#    samples <- new.env()
+#    load(samples.file, envir = samples) ## loads ensemble.samples, trait.samples, sa.samples, runs.samples, env.samples
+#    trait.samples <- samples$trait.samples
+#    ensemble.samples <- samples$ensemble.samples
+#    sa.samples <- samples$sa.samples
+#    runs.samples <- samples$runs.samples
+#    ## env.samples <- samples$env.samples
```

Comment:
I strongly recommend keeping this pattern of loading all Rdata files into their own environment -- not only does it keep the package checks from complaining about variables with no visible binding, it means there's no ambiguity about what names were loaded from which file.

The hack of reading the file into an environment and then immediately assigning back out to separate variables was just to minimize lines changed while quieting the check note.  I'd support a further refactor to use all of these directly from thee loading environment instead (e.g. write all references to `sa.samples` as `samples$sa.samples`)

Reply:
to be honest, this was me trying to resolve a merge conflict and not having time to go back and check the code in more detail (it's been months since I worked on this PR). My plan had been to take a closer look at this function and once I had pulled in the latest code